db/silkworm/common/chain_genesis.hpp

```diff
@@ -0,0 +1,9 @@
9d77d73b1fd0519fc7\":{\"balance\":\"500000000000000000000\"},\"5f68a24c7eb4117667737b33393fb3c2148a53b6\":{\"balance\":\"51800000000000000000\"},\"d8f665fd8cd5c2bcc6ddc0a8ae521e4dc6aa6060\":{\"balance\":\"1700000000000000000000\"},\"d66acc0d11b689cea6d9ea5ff4014c224a5dc7c4\":{\"balance\":\"18200000000000000000\"},\"6e72b2a1186a8e2916543b1cb36a68870ea5d197\":{\"balance\":\"186000000000000000000\"},\"5102a4a42077e11c58df4773e3ac944623a66d9f\":{\"balance\":\"2000325000000000000000\"},\"72480bede81ad96423f2228b5c61be44fb523100\":{\"balance\":\"6400000000000000000000\"},\"e076db30ab486f79194ebbc45d8fab9a9242f654\":{\"balance\":\"4840000000000000000000\"},\"8ceea15eec3bdad8023f98ecf25b2b8fef27db29\":{\"balance\":\"2000000000000000000000\"},\"40652360d6716dc55cf9aab21f3482f816cc2cbd\":{\"balance\":\"10000000000000000000000\"},\"13e02fb448d6c84ae17db310ad286d056160da95\":{\"balance\":\"2000000000000000000000\"},\"d6598b1386e93c5ccb9602ff4bbbecdbd3701dc4\":{\"balance\":\"224096000000000000000\"},\"d5ea472cb9466018110af00c37495b5c2c713112\":{\"balance\":\"4997800000000000000000\"},\"bb75cb5051a0b0944b4673ca752a97037f7c8c15\":{\"balance\":\"200000000000000000000\"},\"8af626a5f327d7506589eeb7010ff9c9446020d2\":{\"balance\":\"1400000000000000000000\"},\"318c76ecfd8af68d70555352e1f601e35988042d\":{\"balance\":\"501600000000000000000\"},\"5c3d19441d196cb443662020fcad7fbb79b29e78\":{\"balance\":\"14300000000000000000\"},\"27101a0f56d39a88c5a84f9b324cdde33e5cb68c\":{\"balance\":\"2000000000000000000000\"},\"e229e746a83f2ce253b0b03eb1472411b57e5700\":{\"balance\":\"5730000000000000000000\"},\"604cdf18628dbfa8329194d478dd5201eecc4be7\":{\"balance\":\"23000000000000000000\"},\"657473774f63ac3d6279fd0743d5790c4f161503\":{\"balance\":\"200000000000000000000\"},\"1ddefefd35ab8f658b2471e54790bc17af98dea4\":{\"balance\":\"1000000000000000000000\"},\"ac3900298dd14d7cc96d4abb428da1bae213ffed\":{\"balance\":\"24730250000000000000000\"},\"944f07b96f90c5f0d7c0c580533149f3f585a078\":{\"balance\":\"74000000000000000000\"},\"232c6d03b5b6e6711efff190e49c28eef36c82b0\":{\"balance\":\"1337000000000000000000\"},\"c87c77e3c24adecdcd1038a38b56e18dead3b702\":{\"balance\":\"8800000000000000000000\"},\"c4b6e5f09cc1b90df07803ce3d4d13766a9c46f4\":{\"balance\":\"6000000000000000000000\"},\"d44334b4e23a169a0c16bd21e866bba52d970587\":{\"balance\":\"2600000000000000000000\"},\"7757a4b9cc3d0247ccaaeb9909a0e56e1dd6dcc2\":{\"balance\":\"20000000000000000000\"},\"cf694081c76d18c64ca71382be5cd63b3cb476f8\":{\"balance\":\"1000000000000000000000\"},\"133e4f15e1e39c53435930aaedf3e0fe56fde843\":{\"balance\":\"20000000000000000000\"},\"f067fb10dfb293e998abe564c055e3348f9fbf1e\":{\"balance\":\"2000000000000000000000\"},\"94449c01b32a7fa55af8104f42cdd844aa8cbc40\":{\"balance\":\"16548000000000000000000\"},\"0e2094ac1654a46ba1c4d3a40bb8c17da7f39688\":{\"balance\":\"358000000000000000000\"},\"738ca94db7ce8be1c3056cd6988eb376359f3353\":{\"balance\":\"25500000000000000000000\"},\"0cfb172335b16c87d519cd1475530
```

Comment:
Why wrap JSON as C++ strings? IMHO separate JSON files for mainnet, Rinkeby, etc would be better for readability.

Reply:
if we want to initialize a mainnet node on a future silkworm client, we should have to pre-made configuration without the user specifying a path (like it's made on tg, geth, etc...)

Reply:
just my two cents on the matter, i just believe that the main genesis should be native and not be inputted by the user

Reply:
OK, fair enough. Could you please then use [raw literals](https://www.geeksforgeeks.org/raw-string-literal-c/) and formatted JSON so that it looks something like
```
std::string kMainnetGenesis = R"({"alloc":
    "3282791d6fd713f1e94f4bfd565eaa78b3a0599d":{"balance":"1337000000000000000000"},
    "17961d633bcf20a7b029a7d94b7df4da2ec5427f":{"balance":"229427000000000000000"},
    ...
})";
```
instead of extremely long lines.

Reply:
Also, you can create nlohmann::json directly:
https://github.com/nlohmann/json#json-as-first-class-data-type
They give an example
```
// instead, you could also write (which looks very similar to the JSON above)
json j2 = {
  {"pi", 3.141},
  {"happy", true},
  {"name", "Niels"},
  {"nothing", nullptr},
  {"answer", {
    {"everything", 42}
  }},
  {"list", {1, 0, 2}},
  {"object", {
    {"currency", "USD"},
    {"value", 42.99}
  }}
};
```

Reply:
sure

Reply:
I think that the last option would be too time expensive (it's a lot of allocs) (26771 lines)

Reply:
now i made it more readable